src/libraries/icubmod/bcbBattery/bcbBattery.cpp

```diff
@@ -285,26 +204,27 @@ void BcbBattery::run()
         sprintf(buff, "battery status: %+6.1fA   % 6.1fV   charge:% 6.1f%%    time: %s", battery_current, battery_voltage, battery_charge, battery_timestamp);
         yDebug("BcbBattery::run() log_buffer is: %s", buff);
     }
+
+    //empty the buffer
+    serial_buff[0] = 0;
```

Comment:
Did you mean to empty all the buffer? I guess that this only set to zero the first element.